### PR TITLE
Pass parent to xblock on load

### PR DESCRIPTION
**Pull request by cpennington**
@doctoryes @alawibaba @utkjad: This should help optimize parent traversal.

### CODE REVIEW

File: xblock/mixins.py [hunk 1]
```diff
@@ -345,6 +352,33 @@ def get_parent(self):
             self._parent_block_id = self.parent
         return self._parent_block
 
+    @property
+    def has_cached_parent(self):
+        """Return whether this block has a cached parent block."""
+        return self.parent and self._parent_block is not None and self._parent_block_id == self.parent
```
Comment (doctoryes) [inline xblock/mixins.py]: Why the middle `is not None`?

File: xblock/runtime.py [hunk 2]
```diff
@@ -628,14 +628,15 @@ def construct_xblock_from_class(self, cls, scope_ids, field_data=None, *args, **
         Construct a new xblock of type cls, mixing in the mixins
         defined for this application.
         """
-        return self.mixologist.mix(cls)(
+        block = self.mixologist.mix(cls)(
```
Comment (doctoryes) [inline xblock/runtime.py]: Was there a subtle reason behind this change?

Comment (cpennington) [inline xblock/runtime.py]: Nope, that was debugging detritus

**Comment by doctoryes** [comment]
:+1:

**Comment by alawibaba** [comment]
:+1:

**Pull request merged**